src/formatter.ts

```diff
@@ -0,0 +1,156 @@
+import { blueBright, dim, red, reset, underline, yellow } from 'chalk';
+import { ESLint } from 'eslint';
+import stripAnsi from 'strip-ansi';
+import table from 'text-table';
+import { TodoResultMessage } from 'types/estlint-type-extension';
+
+function pluralize(word: string, count: number): string {
+  return count === 1 ? word : `${word}s`;
+}
+
+export function formatter(
+  results: ESLint.LintResult[],
+  shouldIncludeTodo?: boolean
+): string {
+  let output = '\n';
+  let errorCount = 0;
+  let warningCount = 0;
+  let todoCount = 0;
+  let fixableErrorCount = 0;
+  let fixableWarningCount = 0;
+  let fixableTodoCount = 0;
+  let chalkColorFunction = yellow;
+  let hasAnyErrors = false;
+
+  results.forEach((result) => {
```

Comment:
Any chance we could split this loop contents up into functions? It's really long/hard to read right now, and I suspect it has a high cyclomatic complexity.

Reply:
Sure I agree. This function was originally created from the default ESLint Formatter and its structure was hardly changed except for the Todo stuff, which I believe ended up increasing its complexity.

I made a big change on this one to make it a lot simpler. There's now the exported `formatter` which is composed of `formatResults`, `formatMessages`, `formatSummary`.